aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -285,8 +285,10 @@ TORCH_META_FUNC(amax)
 (const Tensor& self, IntArrayRef dim, bool keepdim) {
   auto maybe_result = maybe_get_output();
   if (maybe_result.defined()) {
-    TORCH_CHECK(self.scalar_type() == maybe_result.scalar_type(), "Expected the dtype for input and out to match, but got ",
-            self.scalar_type(), " for input's dtype and ",  maybe_result.scalar_type(), " for out's dtype.");
+    TORCH_CHECK(
+      self.scalar_type() == maybe_result.scalar_type(),
+      "Attempting to cast from ", self.scalar_type(), " to out tensor with dtype ", maybe_result.scalar_type(),
+      ", but this can't be cast because it is not safe!");
   }
```

Comment:
I've changed this to use the error message from primTorch (instead of the other way around) because I think the primTorch one is more descriptive.